experimental/torchprep/torchprep/main.py

```diff
@@ -0,0 +1,141 @@
+import typer
+import torch
+from typing import List
+import time
+import os
+from enum import Enum
+from pathlib import Path
+import math
+from tqdm import tqdm
+import torch.fx as fx
+
+
+# TODO: Make commands return a nn.Module instead of None so operations can be composed
+
+app = typer.Typer()
+
+class Precision(Enum):
+    int8 = "int8"
+    float16 = "float16"
+
+class Device(str, Enum):
+    cpu = "cpu"
+    gpu = "gpu"
+
+class Architecture(str, Enum):
+    ipex = "ipex"
+    tensorrt = "tensorrt"
+    fastertransformer = "fastertransformer"
+
+
+@app.command()
+def hello() -> None:
+    typer.echo(f"Hello torchprep")
+
+@app.command()
+def distill(model_path : Path, device : Device, parameter_scaling : int, layer_scaling : int = None, profile : List[int] = None) -> torch.nn.Module:
+    """
+    Create a smaller student model by setting a distillation ratio and teach it how to behave exactly like your existing model
+    """
+    typer.echo(f"Coming soon")
+    typer.echo("See this notebook for more information https://colab.research.google.com/drive/1RzQtprrHx8PokLQsFiQPAKzfn_DiTpDN?usp=sharing")
+
+@app.command()
+def fuse(model_path : Path, device : Device = Device.cpu) -> torch.nn.Module:
+    """
+    Supports optimizations including conv/bn fusion, dropout removal and mkl layout optimizations
+    Requires Pytorch Nightly
+    https://github.com/pytorch/pytorch/blob/master/torch/fx/experimental/optimization.py#L234
+    """
+    model = load_model(model_path, device)
+    optimized_model = torch.fx.experimental.optimization.optimize_for_inference(model)
+
+    torch.save(optimized_model, 'optimized_model.pt') 
+    return optimized_model
+    typer.echo("coming soon")
+
+@app.command()
+def env_variables(model_path : Path, architecture : Architecture) -> None:
+    """
+    Set environment variables for optimized inference. Run this command on the machine where inference will happen!
+    """
+    if architecture == Architecture.ipex:
```

Comment:
How to integrate [ipex parameters](https://github.com/intel/intel-extension-for-pytorch/blob/master/tutorials/intro_launch.md#usage-of-launch-script)?